python/cuml/_thirdparty/sklearn/preprocessing/_imputation.py

```diff
@@ -379,27 +379,40 @@ def _dense_fit(self, X, strategy, missing_values, fill_value):
             count_missing_values = mask.sum(axis=0)
             n_elems = X.shape[0] - count_missing_values
             mean = np.nansum(X, axis=0)
-            mean -= (count_missing_values * missing_values)
+            if not np.isnan(missing_values):
+                mean -= (count_missing_values * missing_values)
             mean /= n_elems
             return mean
 
         # Median
         elif strategy == "median":
+            if X.size == 0:
```

Comment:
why would x be size 0?

Reply:
Before I muck with things too much, @viclafargue, is there a specific reason why we dropped the check for `ensure_min_features` that exists in sklearn from [here](https://github.com/rapidsai/cuml/blob/dd50b750b6756093b682b9ca2ba1f64e2a75d73c/python/cuml/thirdparty_adapters/adapters.py#L160). Just want to make sure I'm not rehashing something that you've already thought about.

@JohnZed I'm tempted to leave this as is and do a follow-up PR to correctly raise a ValueError when imputing on an empty array. If we leave this as is, we still get reasonable behavior. The medians will still get computed as nans, and when we impute on an empty array, we return an empty array. This is not the same behavior as sklearn, but it's not especially alarming either. Removing this check would throw an IndexError at [line 412](https://github.com/rapidsai/cuml/pull/2973/files#diff-6eb3baa332b96d9c2a1a79101b54e7d74c7510d65407a7cb54db9d3606d406f6R412-R413).

Reply:
Okay, great! I'm going to file an issue for this and start working on it as  a separate PR, but feel free to holler @JohnZed if you think this is a must-have as part of this PR.